xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-default/src/main/java/org/xwiki/mentions/internal/async/MentionsThreadFactory.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.mentions.internal.async;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadFactory;
+
+import static java.lang.Thread.NORM_PRIORITY;
+
+/**
+ * {@link ThreadFactory} dedicated to asynchronous tasks needed for the mentions.
+ *
+ * @version $Id$
+ * @since 12.6RC1
+ */
+class MentionsThreadFactory implements ThreadFactory
+{
+    private static final String THREAD_NAME = "Mentions pool thread";
+
+    private final ThreadFactory threadFactory = Executors.defaultThreadFactory();
+
+    @Override
+    public Thread newThread(Runnable runnable)
+    {
+        Thread thread = this.threadFactory.newThread(runnable);
+        thread.setDaemon(true);
```

Comment:
Note that this means the thread is killed right away when Java stops. Is this OK in your use case or do you need to wait for the thread to finish processing the mentions ?

Reply:
Indeed, that's probably better to remove this line. Is it necessary to define a timeout mechanism in case a thread is hanging for too long?

Reply:
For the details I'm not sure since it depends on your need but what you need to have is a way to tell the processing thread that it needs to stop when it's done processing. To trigger this clean stop you need to implement org.xwiki.component.phase.Disposable#dispose() in the component in charge of telling this thread to stop.